### PR TITLE
feat: setup the data schema

### LINKED ISSUE #10: Structure the data-layer
We need to shape the requirements and select the libraries to manage the data layer.

I have already some points learned from the previous discoveries:
- files should have their own struct and should have an unique id assigned
- every entity should have an unique id
- sharing on Automerge works on document level
  - we can implent access checks for the single track files since they live outside Automerge
  - a document sould be our smallest unit to share (e.g. playlist?)

### PR DESCRIPTION
**Pull request by gdorsi**
Changes:
- Move the App component into `ui/pages/Home`
- Create a new Registration component
- Setup an intial schema with [Zod](https://zod.dev/), with the version label for future migrations

Resolves #10

### CODE REVIEW

File: src/lib/filesystem.ts [hunk 1]
```diff
@@ -1,28 +1,13 @@
-// Load the files selected with an input type="file" into the origin
-// private filesystem in order to make them easy to retrieve in the future
-export async function copyToPrivateFileSystem(target: HTMLInputElement) {
-	const files = target.files;
-
-	if (!files) {
-		return [];
-	}
-
+export async function copyToPrivateFileSystem(name: string, file: File) {
```
Comment (gdorsi) [inline src/lib/filesystem.ts:1]: Changed to handle a single file and to get the name as input.

This way we can use the `id` as file name, making possible to store multiple files with the same name.

File: src/state/musicCollection.ts [hunk 2]
```diff
@@ -1,76 +1,44 @@
...
-export type Documents = ReturnType<typeof getDocumentsURLs>;
+import type { MusicCollection, MusicItem } from "./schema";
+import { useUserDocuments } from "./user";
+import { copyToPrivateFileSystem } from "@/lib/filesystem";
 
 export function useMusicCollection() {
-	const repo = useRepo();
-	const documentUrl = useState(() => getDocumentsURLs(repo).musicCollection)[0];
-	const [doc, change] = useDocument<MusicCollectionDocument>(documentUrl);
+	const { collectionsUrls } = useUserDocuments();
```
Comment (gdorsi) [inline src/state/musicCollection.ts:8]: Now we store in a document all the user music collections.

This way in the future we can show the collections list

File: src/ui/components/ui/input.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import * as React from "react";
```
Comment (gdorsi) [inline src/ui/components/ui/input.tsx:1]: Added from shadcdn

File: src/ui/components/ui/label.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import * as React from "react";
```
Comment (gdorsi) [inline src/ui/components/ui/label.tsx:1]: Added from shadcdn

**Review by paolodo — APPROVED** [review]

**Pull request merged**